Update CHANGELOG to mention that @autoclosure forwarding is disabled …

xedin opened the pull request:
…in Swift >= 5

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

xedin committed:
Update CHANGELOG to mention that @autoclosure forwarding is disabled in Swift >= 5

xedin commented:
@AnnaZaks Does this explanation make sense to you?

slavapestov commented on CHANGELOG.md:
```diff
@@ -23,6 +23,22 @@ CHANGELOG
 ---------
+* [SR-9494][]:
+
+  Implicit `@autoclosure` function type forwarding has been disabled,
```
Please add a note that this only happens in -swift-version 5 mode

xedin commented on CHANGELOG.md:
```diff
@@ -23,6 +23,22 @@ CHANGELOG
 ---------
+* [SR-9494][]:
+
+  Starting from `-swift-version 5` implicit `@autoclosure` function type
```
Does this look better, @slavapestov ?

slavapestov commented on CHANGELOG.md:
```diff
@@ -23,6 +23,22 @@ CHANGELOG
+* [SR-9494][]:
+
+  Starting from `-swift-version 5` implicit `@autoclosure` function type
+  forwarding has been disabled and new diagnostic has been added suggesting
```
comma: "Starting from -swift-version
comma: "has been disabled,"
missing 'a': "and a new diagnostic"

slavapestov commented on CHANGELOG.md:
```diff
@@ -23,6 +23,22 @@ CHANGELOG
+
+  Starting from `-swift-version 5` implicit `@autoclosure` function type
+  forwarding has been disabled and new diagnostic has been added suggesting
+  to add `()` to call function type in such case. Call itself would be
```
"to call the function value"
"The call itself"

slavapestov commented on CHANGELOG.md:
```diff
@@ -23,6 +23,22 @@ CHANGELOG
+  Starting from `-swift-version 5` implicit `@autoclosure` function type
+  forwarding has been disabled and new diagnostic has been added suggesting
+  to add `()` to call function type in such case. Call itself would be
+  wrapped in implicit closure and passed to a corresponding parameter.
```
"in an implicit"
"passed to the"

slavapestov reviewed (commented):
Some minor edits

xedin commented:
@slavapestov All fixed, thank you and sorry for completely illiterate.

benrimmington commented:
@xedin You'll need to add a URL to the end of this CHANGELOG.md file.

[SR-9494](https://bugs.swift.org/browse/SR-9494) is probably a duplicate of [SR-5719](https://bugs.swift.org/browse/SR-5719) (also discussed on [forums](https://forums.swift.org/t/how-should-we-fix-sr-5719-for-swift-5/8777)).

xedin commented:
@benrimmington Thank you! I have so many JIRAs by now that I just overlooked that one... I've replaced SR-9494 with SR-5719 (duplicated it in JIRA), and added a link.

xedin commented:
@swift-ci please smoke test

xedin commented:
Going to merge a bit later today unless anybody objects.
